### ISSUE TITLE
.select2-hidden-accessible causes whitespace issue inside scrollable div

### ISSUE THREAD

**Issue by ashgibson** [select2/select2#6168]
Please see jsbin - https://jsbin.com/bobogayire/1/edit?html,js,output

If you scroll down, you will see additional whitespace at the bottom of the page which is caused by the hidden select element.

If I add a style to change .select2-hidden-accessible to display: none, the issue is fixed.

Is there a fix for this?

**Comment by smalos** [comment]
Could you upload a screenshot of what it looks like?

**Comment by eyupfidan** [comment]
There is no problem, please take a screenshot

**Comment by ashgibson** [comment]
The jsbin i have put together shows all the whitespace that is added at the bottom, just scroll down.

**Comment by smalos** [comment]
> Could you upload a screenshot of what it looks like?

Could you upload one? I don't see any whitespace at the bottom.

**Comment by ashgibson** [comment]
2 screenshots below of the jsbin example. 1st screenshot shows when the page is loaded, 2nd after i scroll down showing all the added whitespace.

<img width="1007" alt="Screen Shot 2022-08-30 at 8 59 47 am" src="https://user-images.githubusercontent.com/1744544/187313431-a5793632-7f95-47ac-893e-89898ffe0df8.png">

<img width="999" alt="Screen Shot 2022-08-30 at 8 59 58 am" src="https://user-images.githubusercontent.com/1744544/187313441-e323731d-168b-4880-8290-845fadcd042a.png">

**Comment by smalos** [comment]
This is what it looks like for me.

![select2_scroll](https://user-images.githubusercontent.com/45562597/187350091-de5f859e-07cf-4a0f-b8c8-eab047153b1a.gif)

**Comment by ashgibson** [comment]
Yes that is the same for me. All that extra white space is being added. In Firefox the issue doesn't appear to exist but in Chrome and Safari on Mac it does.

**Comment by imPrashik** [comment]
Fix:

.select2-hidden-accessible{
	position: fixed !important;
}

**Comment by imPrashik** [comment]
Fix:

.select2-hidden-accessible{
	position: fixed !important;
}

**Comment by MariaMozgunova** [comment]
I had to use `select.select2-hidden-accessible` selector
```css
select.select2-hidden-accessible {
    position: fixed !important;
}
```
to increase the specificity of the CSS selector. Otherwise this would have been overriden by the select2 CSS styles.

**Comment by MaxenceMouchard** [comment]
I think it is preferable to place the `<select>` inside a parent container with `position: relative`, in order to avoid the `<select>` in `position: absolute` being positioned relative to a distant ancestor in the DOM, which could be scrollable. In that case, the element may end up rendered in an unexpected place on the page, leading to incorrect positioning and potentially artificially extending the scrollable area, which can cause double scrolling or unexpected overflow and scroll behavior.

In my opinion, this is safer than using `position: fixed`, because fixed completely removes the element from its container context and can break expected positioning and recalculation behavior, especially in scrollable layouts or modals. 

This may create misalignment between the field and its dropdown, particularly during scrolling or resizing.
However, I am not entirely certain about this last point, as I do not fully master all internal behaviors of Select2 regarding how it handles positioning and dropdown rendering.
